### DISCUSSION TITLE
Merge queue Allow specified actors to bypass required pull requests

### DISCUSSION THREAD

**Discussion by elia-bracci-hs** [community/community#45208] [Pull Requests]
### Select Topic Area

Bug

### Body

Hi,

as written in the title I faced that when merge queue is enabled on a specific protected branch, setting `Allow specified actors to bypass required pull requests` doesn't work as expected: actors/users added in that setting list still not able to push on related protected branch.

Did anyone face the some issue or is this intended behavior??

Thank you

**Comment by yesh4gvm** [comment]
Did you find any solution @EliaBracciSumo ? I am facing same issue

**Reply by elia-bracci-hs** [reply]
No 👎

**Comment by lucas-nelson** [comment]
I've run into the same problem. Enabling "Require merge queue" seems to require a pull request for everything regardless of the pull request bypass users setting. It _feels like_ the "Require merge queue" feature needs its own "Allow specified actors to bypass" option created.

For now, we've had to disable the merge queue feature so our existing dependency management workflow can function.

**Comment by ifbarrera** [comment]
We've experienced the same problem. Posted more details in [this closed discussion](https://github.com/orgs/community/discussions/46757#discussioncomment-7051048), has anyone found a workaround?

**Comment by yesh4gvm** [comment]
Well we have found a work around on this  when both the configurations are enabled.

- Require approvals
- Allow specified actors to bypass required pull request

Here is the steps of solutions to overcome.

1. Firstly you required a person to approve the PR.
2. Dividing this into smaller points ( this only works if you have Resolve Merge Conflict (RMC))
i. If you are authorised user to merge PR and still gives error to merge into protected branch the way around is to create new branch.
ii. After creation of new branch merge changes into new branch
iii. Now push the changes of new branch to the branch we need to merge before.

Example of above with real life problem.

- A is preprod branch 
- B is feature/Xxxx
- C is master branch
- D is new branch of A

- A is already updated by C.
- B needs to be merge in A and C.
- When we try to merge B in A it may some times give error of RMC.
- And we resolve RMC on our terminal and try to push we are unable to push.
- The reason will be you are not authorised person to push to protected branch. ( similar kinda errror)
- Now you will need to create new branch from A.
- After creation of new branch D you will need to merge your changes of B.
- After merging and pushing the code to remote create PR of D branch pushing to A branch.
- This is will not show RMC and you will be easliy merge the code directly from Github.
- After merging code from github you can delete new branch D from local and remote.

I hope you guys @EliaBracciSumo @lucas-nelson @ifbarrera and others in future this will help incase of my scenarios.

**Comment by fallemand** [comment]
I'm facing the same issue. 
We have a bot with `Allow specified actors to bypass required to pull requests,` but it's not allowed to add to the queue, given it requires approval from the CODEOWNERS: `Pull request Waiting on code owner review from getyourguide/dex-sdui and/or getyourguide/rnb.` 
Since the account has bypass rights, it should be allowed to add PRs to the queue.

**Comment by theago-ls** [comment]
**ONE POSSIBLE SOLUTION:**

Our repo had branch protection rules in Settings > Branches, and rulesets for the same branch in Settings > Ruleset. Both had bypass permission and bypass list. Removing the rule from Branches fixed the problem for us.

GitHub has a warning about using the new ruleset instead of the classic branch protection rules, but it should be added that using both at the same time can result in conflicts (our case)
